src/renderer/services/neo.js

```diff
@@ -555,113 +544,113 @@ export default {
-                        holdingBalance.marketCap = val[`market_cap_${lowercaseCurrency}`];
-                        holdingBalance.change24hrPercent = val.percent_change_24h;
-                        holdingBalance.unitValue = val[`price_${lowercaseCurrency}`]
-                          ? parseFloat(val[`price_${lowercaseCurrency}`]) : 0;
-                        holdingBalance.unitValue24hrAgo = holdingBalance.unitValue
-                          / (1 + (holdingBalance.change24hrPercent / 100.0));
-                        holdingBalance.change24hrValue = (holdingBalance.unitValue * holdingBalance.balance)
-                          - (holdingBalance.unitValue24hrAgo * holdingBalance.balance);
-                        holdingBalance.totalValue = holdingBalance.unitValue * holdingBalance.balance;
-                        if (holdingBalance.unitValue === null || isNaN(holdingBalance.unitValue)) {
-                          holdingBalance.totalValue = null;
-                          holdingBalance.change24hrPercent = null;
-                          holdingBalance.change24hrValue = null;
-                        }
-                        done();
+                  promises.push(dex.fetchContractBalance(holding.assetId)
+                    .then((res) => {
+                      holding.contractBalance = toBigNumber(res);
+                      holding.availableBalance = calculateHoldingAvailableBalance(holding);
+                      holding.totalBalance = calculateHoldingTotalBalance(holding);
+                    })
+                    .catch((e) => {
+                      const existingHolding = this.getHolding(holding.assetId);
+                      if (existingHolding) {
+                        holding.contractBalance = existingHolding.contractBalance;
+                        holding.availableBalance = calculateHoldingAvailableBalance(holding);
+                        holding.totalBalance = calculateHoldingTotalBalance(holding);
+                      }
+                      alerts.networkException(e);
+                    }));
+
+                  promises.push(dex.fetchOpenOrderBalance(holding.assetId)
+                    .then((res) => {
+                      holding.openOrdersBalance = toBigNumber(res);
+                      holding.availableBalance = calculateHoldingAvailableBalance(holding);
+                      holding.totalBalance = calculateHoldingTotalBalance(holding);
+                    })
+                    .catch((e) => {
+                      alerts.networkException(e);
+                      const existingHolding = this.getHolding(holding.assetId);
+                      if (existingHolding) {
+                        holding.openOrdersBalance = existingHolding.openOrdersBalance;
+                        holding.availableBalance = calculateHoldingAvailableBalance(holding);
+                        holding.totalBalance = calculateHoldingTotalBalance(holding);
+                      }
+                    }));
```

Comment:
These two promise pushes are so similar. There's I think 1 or 2 differences between them. I wonder if you can extrapolate them so we don't repeat the code. Not a big deal, but it would be cool..

Reply:
Done.